src/main/java/viper/Duke.java

```diff
@@ -0,0 +1,47 @@
+package viper;
+
+import commands.Command;
+import exceptions.DukeException;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.text.ParseException;
+
+public class Duke {
+    
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) throws IOException {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.loadTasks());
+        } catch (FileNotFoundException | DukeException | ParseException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
```

Comment:
Would renaming boolean isRunning = true, followed by while (isRunning) be more intuitive?